docfx_yaml/extension.py

```diff
@@ -939,56 +939,77 @@ def _update_friendly_package_name(path):
         if _type in [METHOD, FUNCTION, CLASS]:
             argspec = inspect.getfullargspec(obj) # noqa
             type_map = {}
-            if argspec.annotations:
-                for annotation in argspec.annotations:
-                    if annotation == "return":
-                        continue
-                    try:
-                        type_map[annotation] = _extract_type_name(
-                            argspec.annotations[annotation])
-                    except AttributeError:
-                        print(f"Could not parse argument information for {annotation}.")
-                        continue
-
-            # Add up the number of arguments. `argspec.args` contains a list of
-            # all the arguments from the function.
-            arg_count += len(argspec.args)
-            for arg in argspec.args:
-                arg_map = {}
-                # Ignore adding in entry for "self"
-                if arg != 'cls':
-                    arg_map['id'] = arg
-                    if arg in type_map:
-                        arg_map['var_type'] = type_map[arg]
-                        args.append(arg_map)
+
+            annotations = getattr(argspec, 'annotations', [])
```

Comment:
Does this need tests?

Reply:
Normally the test would involve updating the goldens as this function is called directly from Sphinx and can't be mocked easily.

I could refactor the logic out and add unit tests separately as a followup, and release only when the refactoring of this method is done.